lib/path/default.nix

```
@@ -94,6 +95,47 @@ let
 
 in /* No rec! Add dependencies on this file at the top. */ {
 
+  /* Append a subpath string to an absolute path value, the result is an absolute path value.
+
+    Like `path + ("/" + string)` but safer, because it errors for potentially unexpected results.
+    More specifically, it checks that the first argument is a path value,
```

```suggestion
    More specifically, it checks that the first argument is a [path],
```

Link to the Nix manual to make clear what we mean.
Why do we say "path value"? Is there something specific about the value?

---

I called it "path value" because "path" is a very ambiguous term. With "value" it becomes much clearer that this is a Nix type. But I guess if we link to it that takes care of it too.